tutor/src/screens/question-library/exercises-display.jsx

```diff
@@ -106,6 +110,28 @@ class ExercisesDisplay extends React.Component {
 
   scroller = new Scroller({ windowImpl: this.windowImpl });
 
+  componentWillUnmount() {
+    this.props.exercises.clear();
+    document.removeEventListener('scroll', this.checkIsControlsSticky);
+  }
+
+  componentDidMount() {
+    document.addEventListener('scroll', this.checkIsControlsSticky);
+  }
+
+  // Checking if the controls-bar is sticked to the top.
+  // If so, show box-shadow.
+  @action.bound checkIsControlsSticky() {
+    const windowImpl = this.windowImpl || window;
```

Comment:
windowImpl should be from props, however it won't be needed if you use windowScroll.

Instead how about this:
```js

class ExercisesDisplay extends React.Component {
...

windowScroll = new WindowScroll(this.props.windowImpl)

@computed areControlsSticky() {
  const topOffset = this.showingDetails ? 160 : 250;
   return this.windowScroll.y >=  topOffset;
}
```

Reply:
this works! 👍